src/collision_node.cpp

```
@@ -53,13 +53,16 @@ void collide(CollisionTraversalNodeBase* node, const CollisionRequest& request,
     else
       collisionNonRecurse(node, front_list, sqrDistLowerBound);
 
+    const FCL_REAL dummy_precision =
+        std::sqrt(Eigen::NumTraits<FCL_REAL>::epsilon());
+    HPP_FCL_UNUSED_VARIABLE(dummy_precision);
     if (!std::isnan(sqrDistLowerBound)) {
       if (sqrDistLowerBound == 0) {
-        assert(result.distance_lower_bound <= 0);
+        assert(result.distance_lower_bound <= dummy_precision);
       } else {
         assert(result.distance_lower_bound * result.distance_lower_bound -
                    sqrDistLowerBound <
-               1e-8);
+               dummy_precision);
```

This code is checking if the result is correct.

1. Are you sure throwing an exception is not a better behavior ?
2. I would put this code in a separated static function (for the symbol visibility) to split the useful code from the error checking.

---

I think you're right, this should probably throw an exception.

Contrary to the assertions in GJK for which GJK has a fallback behavior and will still work fine, here if the asserts are triggered it means the entire collide call is wrong.
So throwing an exception is probably better.

---

Got update from @jcarpent.
This is deep in the code and the throw will slow things down. I will keep the function but replace the throws by asserts.